packages/test/test-end-to-end-tests/src/test/createNewSummaryCachingTests.spec.ts

```diff
@@ -0,0 +1,92 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import {
+    ContainerRuntimeFactoryWithDefaultDataStore,
+    DataObject,
+    DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { ISummaryConfiguration } from "@fluidframework/protocol-definitions";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { MockLogger } from "@fluidframework/test-runtime-utils";
+import { describeNoCompat } from "@fluidframework/test-version-utils";
+import { IContainerRuntimeOptions } from "@fluidframework/container-runtime";
+import { flattenRuntimeOptions } from "./flattenRuntimeOptions";
+
+class TestDataObject extends DataObject {
+    public get _root() {
+        return this.root;
+    }
+    public get _context() {
+        return this.context;
+    }
+}
+
+describeNoCompat("Cache CreateNewSummary", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    const dataObjectFactory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+
+    const IdleDetectionTime = 100;
+    const summaryConfigOverrides: Partial<ISummaryConfiguration> = {
+        idleTime: IdleDetectionTime,
+        maxTime: IdleDetectionTime * 12,
+    };
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            generateSummaries: true,
+            initialSummarizerDelayMs: 10,
+            summaryConfigOverrides,
+        },
+        gcOptions: {
+            gcAllowed: true,
+        },
+    };
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        undefined,
+        flattenRuntimeOptions(runtimeOptions),
+    );
+
+    let mockLogger: MockLogger;
+
+    beforeEach(function() {
+        provider = getTestObjectProvider();
+        // Currently, only ODSP caches new summary.
+        if (provider.driver.type !== "odsp") {
+            this.skip();
+        }
+    });
+
+    it("should fetch from cache when second client loads the container", async () => {
+        mockLogger = new MockLogger();
+        // Create a Container for the first client.
+        const mainContainer = await provider.createContainer(runtimeFactory, { logger: mockLogger });
+        assert.strictEqual(mainContainer.attachState, "Attached", "container was not attached");
+        const mainDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        await dataObjectFactory.createInstance(mainDataStore._context.containerRuntime);
```

Comment:
Is there a reason for creating a second data store?

Reply:
yeah i think Jatin said to request more than one data stores and make sure they are loaded. I haven't figured out how to request the 2nd data store yet😅

Reply:
@jatgarg Does creating a second data store add any value?

@chensixx To request the new data store, you will have to store its handle in the root DDS of mainDataStore. Then in the second container, you should get the handle from the root DDS and call get() on it.

Reply:
@agarwal-navin Actually just making sure that all datastore/dds are loaded properly. Storing handle and getting will add extra test that ok dds are also loaded fine with contents.